src/inference/elbo.ad.js

```diff
@@ -303,7 +303,12 @@ module.exports = function(env) {
       return k(s);
     },
 
-    mapDataFetch: function(data, batchSize, address) {
+    mapDataFetch: function(data, opts, address) {
+
+      var batchSize = opts.batchSize !== undefined ? opts.batchSize : data.length;
+      if (batchSize < 0 || batchSize > data.length) {
```

Comment:
I'm not sure. One argument for not doing so might be something like by setting it to zero we're not only specifying something about how the inference strategy, we changing the objective (by effectively removing the `mapData`). On the other hand, if there's utility in having such a thing I'd be happy to add it. Separate PR?